### ISSUE TITLE
Bring back `CheckoutSdkBuilder.uri` as a builder parameter

### ISSUE THREAD

**Issue by kdrakon** [checkout/checkout-sdk-java#268]
### Environment

* Checkout SDK version: 6.0.2

### Description

`CheckoutSdkBuilder` no longer allows you to specify a `uri` for the API URI that the client should communicate with. It's now inferred by the `Environment` enum parameter. Although this ensures the client correctly connects to a valid checkout.com API URI, it limits integrators to use services like [MockLab](https://get.mocklab.io/) or anything else that can simulate API responses. In CI/CD environment, this means in many cases, integrators are forced to communicate with https://api.sandbox.checkout.com

## Proposed Solution

Bring back the ability to provide a URI as an override to the one provided on `com.checkout.Environment`

**Comment by Armando-Ibarra** [comment]
Hey @kdrakon  sorry for coming late on this.
Question isn't the httpClient parameter enough to Proxy your calls?

like

` CheckoutSdk.builder()
.staticKeys()
.secretKey("secret")
.environment(Environment.SANDBOX)
.httpClientBuilder(HttpClientBuilder
  .create()
  .setProxy(HttpHost.create("http:fake"))
  .setConnectionTimeToLive(3, TimeUnit.SECONDS))
.build();`

I've considering supporting somehow something like "service mesh", but I thought that doing through custom http client will be enough, otherwise I think we will need to add some Customizations to Environment enum .

**Comment by aris-hubpay** [comment]
HI, this is an issue for our integration as well, I tried to use what you suggested here, however whenever I setup this `httpClientBuilder(HttpClientBuilder.create()
                .setProxy(HttpHost.create("https://api.sandbox.checkout.com")))` with the sandbox endoint, I am getting the following response 
```
2022-11-09 15:34:11.112  INFO 71407 --- [onPool-worker-1] com.checkout.ApacheHttpClientTransport   : POST: https://api.sandbox.checkout.com/payments
2022-11-09 15:34:11.112  INFO 71407 --- [onPool-worker-1] com.checkout.ApacheHttpClientTransport   : Request: [user-agent: checkout-sdk-java/6.0.6, Accept: application/json;charset=UTF-8]
2022-11-09 15:34:11.449  INFO 71407 --- [onPool-worker-1] com.checkout.ApacheHttpClientTransport   : Response: 404 [Date: Wed, 09 Nov 2022 13:34:11 GMT, Connection: close, Content-Length: 0]
2022-11-09 15:34:11.450 ERROR 71407 --- [nio-3000-exec-4] o.a.c.c.C.[.[.[/].[dispatcherServlet]    : Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processing failed; nested exception is java.lang.RuntimeException: Cannot get response from checkout create payment] with root cause

com.checkout.CheckoutApiException: The API response status code (404) does not indicate success.
```
If I don't supply the `httpClientBuilder` it works as expected.

**Comment by Armando-Ibarra** [comment]
Hi @aris-hubpay.
Ok we'll work on this, I'll circle back soon with a formal solution.

**Comment by Armando-Ibarra** [comment]
Hi @kdrakon @aris-hubpay 

Sorry to came too late with a custom solution for this, given the existing usage that Environment enum have, we choose to implement a custom Class to provide custom URI's, here is the PR. I think this is the easiest solution, and you don't really need to implement a lot of stuff.

Here is the PR #288 

If you have any comments please let us know, next week we might release this.

**Comment by aris-hubpay** [comment]
Thank you for looking into that. 
When do you think the new version of the sdk that included the fix will be released?

**Comment by Armando-Ibarra** [comment]
@aris-hubpay I'm going to release this next week, probably Monday so that I can unblock you

**Comment by Armando-Ibarra** [comment]
@aris-hubpay @kdrakon We just released #292 a new version of the SDK which includes the custom Environment functionality, details were added to the README, let us know if you face any issues.
Thanks!
